code/modules/religion/pyre_rites.dm

```diff
@@ -0,0 +1,151 @@
+			to_chat(user, span_warning("Only carbon lifeforms can be properly burned for the sacrifice!"))
+			return FALSE
+		chosen_sacrifice = corpse
+		if(chosen_sacrifice.stat != DEAD)
+			to_chat(user, span_warning("You can only sacrifice dead bodies, this one is still alive!"))
+			return FALSE
+		if(!chosen_sacrifice.on_fire)
+			to_chat(user, span_warning("This corpse needs to be on fire to be sacrificed!"))
+			return FALSE
+		return ..()
+
+/datum/religion_rites/burning_sacrifice/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	..()
+	if(!(chosen_sacrifice in religious_tool.buckled_mobs)) //checks one last time if the right corpse is still buckled
+		to_chat(user, span_warning("The right sacrifice is no longer on the altar!"))
+		chosen_sacrifice = null
+		return FALSE
+	if(!chosen_sacrifice.on_fire)
+		to_chat(user, span_warning("The sacrifice is no longer on fire, it needs to burn until the end of the rite!"))
+		chosen_sacrifice = null
+		return FALSE
+	if(chosen_sacrifice.stat != DEAD)
+		to_chat(user, span_warning("The sacrifice has to stay dead for the rite to work!"))
+		chosen_sacrifice = null
+		return FALSE
+	var/favor_gained = 100 + round(chosen_sacrifice.getFireLoss())
+	GLOB.religious_sect.adjust_favor(favor_gained, user)
+	to_chat(user, span_notice("[GLOB.deity] absorbs the burning corpse and any trace of fire with it. [GLOB.deity] rewards you with [favor_gained] favor."))
+	chosen_sacrifice.dust(force = TRUE)
+	playsound(get_turf(religious_tool), 'sound/effects/supermatter.ogg', 50, TRUE)
+	chosen_sacrifice = null
+	return TRUE
+
+/datum/religion_rites/infinite_candle
+	name = "Immortal Candles"
+	desc = "Creates 5 candles that never run out of wax."
+	ritual_length = 10 SECONDS
+	invoke_msg = "Burn bright, little candles, for you will only extinguish along with the universe."
+	favor_cost = 200
+
+/datum/religion_rites/infinite_candle/invoke_effect(mob/living/user, atom/movable/religious_tool)
+	..()
+	var/altar_turf = get_turf(religious_tool)
+	for(var/i in 1 to 5)
+		new /obj/item/flashlight/flare/candle/infinite(altar_turf)
+	playsound(altar_turf, 'sound/magic/fireball.ogg', 50, TRUE)
+	return TRUE
+
+/datum/religion_rites/blazing_star
+	name = "Blazing Star"
+	desc = "Enchants a holy arrow to set someone on fire on hit, or if the victim is already on fire... note, this consumes the arrow."
+	ritual_length = 15 SECONDS
+	ritual_invocations = list(
+		"And so to keep the Ever-Burning candle protected ...",
+		"... grant this feeble bolt your blessing ...",
+		"... make it burn bright ...",
+	)
+	invoke_msg = "... a blazing star is born!"
+	favor_cost = 2000
+	///arrow to enchant
+	var/obj/item/ammo_casing/caseless/arrow/holy/enchant_target
+
+/datum/religion_rites/blazing_star/perform_rite(mob/living/user, atom/religious_tool)
+	for(var/obj/item/ammo_casing/caseless/arrow/holy/can_enchant in get_turf(religious_tool))
+		if(istype(can_enchant, /obj/item/ammo_casing/caseless/arrow/holy/blazing))
```

Comment:
hmm? does this check need to exist? can't you just do the typecheck in the line above this